components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint/src/main/webapp/basicauth.jsp

```diff
@@ -160,42 +164,40 @@
                 identityMgtEndpointContext = IdentityUtil.getServerURL("/accountrecoveryendpoint", true, true);
             }
 
-            URL url = null;
-            HttpURLConnection httpURLConnection = null;
+            HttpClient client = new HttpClient();
+            if ("https".equals(scheme)) {
```

Comment:
Can't we simply instantiate the http client?
Wouldn't the http client packed by kernel take care of registering the correct protocol factory etc.?